Add "crates.io: development update" blogpost (#1471)
rust-lang/blog.rust-lang.org: posts/2025-02-05-crates-io-development-update.md

Please note that this is still considered work-in-progress and e.g. the stability guarantees for the endpoints are not written down and the response schemas are also not fully documented yet.

You can view the OpenAPI description in e.g. a Swagger UI at <https://petstore.swagger.io/> by putting `https://crates.io/api/openapi.json` in the top input field. We decided to not ship a viewer ourselves for now due to security concerns with running it on the same domain as crates.io itself. We may reconsider whether to offer it on a dedicated subdomain in the future if there is enough interest.

![Swagger UI Screenshot](../../../images/2025-02-05-crates-io-development-update/swagger-ui.png)

The OpenAPI description is generated by the [utoipa](https://github.com/juhaku/utoipa) crate, which is a tool that can be integrated with the [axum](https://github.com/tokio-rs/axum) web framework to automatically generate OpenAPI descriptions for all of your endpoints. We would like to thank [Juha Kukkonen](https://github.com/juhaku) for his great work on this tool!

## Support form and "Report Crate" button

> Since the crates.io team is small and mostly consists of volunteers, we do not have the capacity to manually monitor all publishes. Instead, we rely on you, the Rust community, to help us catch malicious crates and users. To make it easier for you to report suspicious crates, we added a "Report Crate" button to all the crate pages. If you come across a crate that you think is malicious or violates the [code of conduct](https://www.rust-lang.org/policies/code-of-conduct) or our [usage policy](https://crates.io/policies), you can now click the "Report Crate" button and fill out the form that appears. This will send an email to the crates.io team, who will then review the crate and take appropriate action if necessary. Thank you to crates.io team member [@eth3lbert](https://github.com/eth3lbert) how worked on the majority of this.

Reviewer: ```suggestion
Since the crates.io team is small and mostly consists of volunteers, we do not have the capacity to manually monitor all publishes. Instead, we rely on you, the Rust community, to help us catch malicious crates and users. To make it easier for you to report suspicious crates, we added a "Report Crate" button to all the crate pages. If you come across a crate that you think is malicious or violates the [code of conduct](https://www.rust-lang.org/policies/code-of-conduct) or our [usage policy](https://crates.io/policies), you can now click the "Report Crate" button and fill out the form that appears. This will send an email to the crates.io team, who will then review the crate and take appropriate action if necessary. Thank you to crates.io team member [@eth3lbert](https://github.com/eth3lbert) who worked on the majority of this.
```

Author: good catch, thanks :)

If you have any issues with the support form or the "Report Crate" button, please let us know. You can also always email us directly at [help@crates.io](mailto:help@crates.io) if you prefer not to use the form.

## Publish notifications

We have added a new feature that allows you to receive email notifications when a new version of your crate is published. This can be useful in detecting unauthorized publishes of your crate or simply to keep track of publishes from other members of your team.

![Publish Notification Screenshot](../../../images/2025-02-05-crates-io-development-update/publish-notification.png)

> This feature was another [long-standing feature request](https://github.com/rust-lang/crates.io/issues/815) from our community, and we were happy to finally implement it. After some initial pushback from a few vocal community members we also implemented a way to opt out of these notifications. If you feel that your account is secure enough, then you can go to your account settings on crates.io and disable these notifications.

Reviewer 2: ```suggestion
This feature was another [long-standing feature request](https://github.com/rust-lang/crates.io/issues/815) from our community, and we were happy to finally implement it. If you'd prefer not to receive publish notifications, then you can go to your account settings on crates.io and disable these notifications.
```

The current sentence feels a bit passive aggressive :sweat_smile:
